tests/api_connexion/endpoints/test_task_instance_endpoint.py

```diff
@@ -1491,3 +1492,111 @@ def test_should_raise_400_for_naive_and_bad_datetime(self, payload, expected, se
         )
         assert response.status_code == 400
         assert response.json['detail'] == expected
+
+
+class _ForceHeartbeatCeleryExecutor(CeleryExecutor):
+    def heartbeat(self):
+        return True
+
+
+@mock.patch(
+    "airflow.executors.executor_loader.ExecutorLoader.get_default_executor",
+    return_value=_ForceHeartbeatCeleryExecutor(),
+)
+class TestTaskInstanceRunEndpoint(TestTaskInstanceEndpoint):
```

Comment:
We should also test for permissions

Reply:
Thanks, this API uses same permissions as setting task instance state. I have added similar tests.